src/dataflow/removed_packages.rs

```diff
@@ -0,0 +1,81 @@
+use crate::dataflow::normalize_global_namespace_package_name;
+use std::borrow::Cow;
+use std::collections::hash_set::HashSet;
+
+/// Holds packages that are removed on the cli
+#[derive(Debug, Default)]
+pub struct RemovedPackages<'a> {
+    pub packages: HashSet<Cow<'a, str>>,
+}
+
+impl<'a> RemovedPackages<'a> {
+    pub fn new_from_package_names(removed_packages: Vec<&'a str>) -> Self {
+        let packages = removed_packages
+            .into_iter()
```

Comment:
I'm not as familiar with `into_iter`, I think of `drain(..)` when I'm using FP-style and destroying one collection and making a new one